geode-core/src/main/java/org/apache/geode/internal/cache/wan/GatewaySenderEventImpl.java

```diff
@@ -710,7 +710,7 @@ public int getDSFID() {
   @Override
   public void toData(DataOutput out,
       SerializationContext context) throws IOException {
-    toDataPre_GEODE_1_13_0_0(out, context);
+    toDataPre_GEODE_1_14_0_0(out, context);
     boolean hasTransaction = this.transactionId != null;
```

Comment:
The version in toData() is 0. So it will not help. I added "if (version == 0 || version >= KnownVersion.GEODE_1_14_0.ordinal()) {" too. The result is the same. It did not improve on reproducing 49196.

Reply:
There is no need for condition check here. The appropriate version of `toData()` is invoked by  `DSFIDSerializerImpl.invokeToData()` and `InternalDataSerializer.invokeToData()`. Similarly, there is no need from condition check in `fromData()`.